app/src/main/java/com/kylecombes/micrepeater/ui/main/VoiceAmplifierFragment.java

```diff
@@ -58,6 +79,32 @@ public View onCreateView(
         mStartStopButton = root.findViewById(R.id.amplifying_control_start_stop_button);
         mStartStopButton.setOnClickListener(startStopButtonClickedListener);
         mStatusImageView = root.findViewById(R.id.voice_amplifier_status_iv);
+        dropdown = root.findViewById(R.id.audio_output_dropdown);
+        //create an adapter to describe how the items are displayed, adapters are used in several places in android.
+        //There are multiple variations of this, but this is the basic variant.
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(getActivity().getBaseContext(),  R.array.audio_output_options, R.layout.spinner_item);
+        //set the spinners adapter to the previously created one.
+        adapter.setDropDownViewResource(R.layout.simple_spinner_dropdown_item);
+        dropdown.setAdapter(adapter);
+        dropdown.setSelection(current_output);
+        dropdown.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                if (position == 0){
+                    audioOutput = "voice";
+                } else if (position == 1){
+                    audioOutput = "alarm";
+                } else if (position == 2) {
+                    audioOutput = "music";
+                }
```

Comment:
Another option here is to have an array with these values at the top of the file:
```java
private static final String[] OUTPUTS = {"voice", "alarm", "music"};
```
Then this code could just be the following:
```java
audioOutput = OUTPUTS[position];
```

Reply:
Sadly, Java has no `indexOf` function for arrays, so we can't just replace the code up above with `currentOutput = Array.indexOf(OUTPUTS, output)` (or whatever the syntax would be).

Reply:
I don't think `audioOutput` needs to be a class variable either. It's just used here and doesn't need to persist between function calls, right?